### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Dongmo12**

### CODE REVIEW

File: enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        num
+    end
+
+    def my_count
+        c = 0
+        self.my_each do |x|
+            block_given? ? (c += 1 if yield(x)) : c += 1
+        end
+        c
+    end
```
Comment (maelfosso) [inline enumerable.rb:55]: Not working. 
```
>> ary = [1, 2, 4, 2]
=> [1, 2, 4, 2]
>> ary.my_count(2)
Traceback (most recent call last):
        7: from /home/maelfosso/.rvm/gems/ruby-2.6.0/bin/ruby_executable_hooks:24:in `<main>'
        6: from /home/maelfosso/.rvm/gems/ruby-2.6.0/bin/ruby_executable_hooks:24:in `eval'
        5: from /usr/share/rvm/rubies/ruby-2.6.0/bin/irb:23:in `<main>'
        4: from /usr/share/rvm/rubies/ruby-2.6.0/bin/irb:23:in `load'
        3: from /usr/share/rvm/rubies/ruby-2.6.0/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):8
        1: from enumerable.rb:46:in `my_count'
ArgumentError (wrong number of arguments (given 1, expected 0))
```

File: enumerable.rb [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    def my_map
+        array = []
+        self.my_each do |x|
+            yield(x) ? (array << true) : (array << false)
+        end
+        array
+    end
```
Comment (maelfosso) [inline enumerable.rb:63]: Not working
```
>> ary = [1, 2, 4, 2]
=> [1, 2, 4, 2]
>> ary.map { |i| i*i }
=> [1, 4, 16, 4]
>> ary.my_map { |i| i*i }
=> [true, true, true, true]
```

File: enumerable.rb [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        
+    def my_inject (val = self[0]) 
+        object = val
+        self.my_each do |x|
+            object = yield(val,x)
+            val = object 
+        end
+        object
+    end
```
Comment (maelfosso) [inline enumerable.rb:73]: Not working
```
>> ary = [1, 2, 4, 2]
=> [1, 2, 4, 2]
>> ary.inject { |sum, n| sum + n } 
=> 9
>> ary.my_inject { |sum, n| sum + n } 
=> 10
>> 
```

File: enumerable.rb [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        self.my_each{|x| select_arr << x if yield(x)}
+        select_arr
+    end
+
+    def my_all?
+        self.my_each do |x|
+        return true unless block_given?
+        return false unless yield(x)
+        end
+    end
```
Comment (maelfosso) [inline enumerable.rb:29]: Not working
```
>> %w[ant bear cat].my_all? { |word| word.length >= 3 }
=> 0...3
>> %w[ant bear cat].all? { |word| word.length >= 3 }
=> true
```

File: enumerable.rb [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    
+    def my_any?
+        m = false
+        self.my_each do |x|
+            (m = true) if yield(x)
+        end
+        m
+    end
```
Comment (maelfosso) [inline enumerable.rb:36]: Not working
```
>> %w[ant bear cat].any?
=> true
>> %w[ant bear cat].my_any?
Traceback (most recent call last):
       11: from /home/maelfosso/.rvm/gems/ruby-2.6.0/bin/ruby_executable_hooks:24:in `<main>'
       10: from /home/maelfosso/.rvm/gems/ruby-2.6.0/bin/ruby_executable_hooks:24:in `eval'
        9: from /usr/share/rvm/rubies/ruby-2.6.0/bin/irb:23:in `<main>'
        8: from /usr/share/rvm/rubies/ruby-2.6.0/bin/irb:23:in `load'
        7: from /usr/share/rvm/rubies/ruby-2.6.0/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        6: from (irb):41
        5: from enumerable.rb:34:in `my_any?'
        4: from enumerable.rb:6:in `my_each'
        3: from enumerable.rb:6:in `each'
        2: from enumerable.rb:7:in `block in my_each'
        1: from enumerable.rb:35:in `block in my_any?'
LocalJumpError (no block given (yield))
```

File: enumerable.rb [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+    def my_none?
+        num = true
+        self.my_each{ |x| num = false if yield(x) }
+        num
+    end
```
Comment (maelfosso) [inline enumerable.rb:43]: Not working
```
>> %w[ant bear cat].none?
=> false
>> %w[ant bear cat].my_none?
Traceback (most recent call last):
       11: from /home/maelfosso/.rvm/gems/ruby-2.6.0/bin/ruby_executable_hooks:24:in `<main>'
       10: from /home/maelfosso/.rvm/gems/ruby-2.6.0/bin/ruby_executable_hooks:24:in `eval'
        9: from /usr/share/rvm/rubies/ruby-2.6.0/bin/irb:23:in `<main>'
        8: from /usr/share/rvm/rubies/ruby-2.6.0/bin/irb:23:in `load'
        7: from /usr/share/rvm/rubies/ruby-2.6.0/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        6: from (irb):43
        5: from enumerable.rb:42:in `my_none?'
        4: from enumerable.rb:6:in `my_each'
        3: from enumerable.rb:6:in `each'
        2: from enumerable.rb:7:in `block in my_each'
        1: from enumerable.rb:42:in `block in my_none?'
LocalJumpError (no block given (yield))
```

**Review by maelfosso — CHANGES_REQUESTED** [review]
Hello @Dongmo12 

You did a very nice job.
There are some changes to do.
Please, correct them and submit a new code review request.

Thanks

File: enumerable.rb (hunk 1 above)
Comment (Dongmo12) [reply enumerable.rb:55]: Done here

File: enumerable.rb (hunk 3 above)
Comment (Dongmo12) [reply enumerable.rb:73]: Done!

Comment (Dongmo12) [reply enumerable.rb]: Done

File: enumerable.rb (hunk 4 above)
Comment (Dongmo12) [reply enumerable.rb:29]: I am done here as well!!

File: enumerable.rb [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+        initial = false
+        self.my_each {|elem| yield(elem) ? initial = true : initial = false }
+        initial
+    end
+    
+    def my_any?
+        self.my_each do |e|
+          return true if yield(e)
+        end
+        true
```
Comment (jofer86) [inline enumerable.rb]: This part of your code is not working. You are always returning true no matter what the block evaluates to, it needs a couple of changes there in order to provide the right output, please kindly give your method a good read, the mistake would be quite easy to find.

File: enumerable.rb [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+    end
+    
+
+    def my_none?
+        num = true
+        self.my_each{ |x| num = false if yield(x) }
+        num
+    end
+
+    def my_count(obj = nil)
```
Comment (jofer86) [inline enumerable.rb]: This method is also not working properly, it return whatever number you pass on if you give it as an argument. Also it does not count the number of elements that return a `true` value if a block is given. Take a close look at the way you wrote your method. The line 49 is forcing a return whenever the count is equal to the argument passed, which is not how it suppose to work. And the same goes for line 50 is just returning whatever count is and not counting the elements.

**Review by jofer86 — CHANGES_REQUESTED** [review]
# :wave: :wave: Hello!!! Great Effort!!!!!

## This was an overall great effort, some points to consider though, your `my_any` has a bit of and issue that can be solved by changing its last line, so no biggie there. Please kindly fix that mistake.

## The my_count function does not work at all, it needs some re-thinking cause it is not returning the appropiate results. Please see to it and kindly fix that.

## Once you are done please do request a new code review.

File: enumerable.rb (hunk 8 above)
Comment (Dongmo12) [reply enumerable.rb]: Thank you!!!

File: enumerable.rb (hunk 7 above)
Comment (Dongmo12) [reply enumerable.rb]: Awesome!!!

**Review by bellom — APPROVED** [review]
Hi @Dongmo12,
Great work making all the necessary corrections from the previous review.
**Well done guy, kindly be aware that you are to use `Git Flow` but instead you used `Github Flow`, make sure to use 'Git Flow' and add Linter in the next project - `OOP`.**
Short Explanation Hint: 
`GitHub Flow`: Pull Requests is raise from feature branches to the master.
`Git Flow`: Pull Requests is raise from feature branches to the development.

**_Go ahead and merge :+1:_**

**Pull request merged**